src/main/java/org/openjdk/jextract/impl/Logger.java

```diff
@@ -50,27 +51,58 @@ public Logger(PrintWriter outWriter, PrintWriter errStream) {
         this.errWriter = errStream;
     }
 
+    public void print(ClangException exp) {
+        errWriter.println(String.format("%1$s: %2$s: %3$s",
+            position(exp.position()),
+            exp.isFatal()? "fatal" : "error",
+            exp.spelling()));
+        nErrors++;
```

Comment:
shouldn't we increase nErrors only based on the severity of the clang diagnostic - e.g. if the diagnostic wrapped by `ClangException` is a warning, then we shouldn't increase `nErrors`. That said, I seem to recollect that the logic in `Parser` only issues a `ClangException` if the severity is > warning. I wonder it might make sense (maybe in a separate PR) to investigate whether adding support for clang warnings might make sense too?